exercises/hypospaces-capacity/ex_rnw/sol_hypospaces_capacity_1_22.Rnw

```diff
@@ -0,0 +1,155 @@
+%	
+\end{align*}
+%
+Thus, $\Hspace \cup \Hspace'$ shatters $\{x_1,x_2\}$ and we infer that $VC_1(\Hspace \cup \Hspace')\geq 2.$
+
+Now, let $x_3\in \R$ be another arbitrary point such that (w.l.o.g$.^{1}$) $x_2<x_3.$
+%
+There is no $h \in \Hspace \cup \Hspace'$ such that $(h(x_1),h(x_2),h(x_3))^\top = (1,-1,1)^\top$  holds.
+%
+Indeed, $h$ is either a 
+%
+\begin{itemize}
+%	
+	\item left-open classifier, i.e. $h=h_r$ for some $r\in \R,$ so that $h(x_3)=+1$ implies $h(x_2)=+1,$
+%	
+	\item right-open classifier, i.e. $h=\tilde h_l$ for some $l\in \R,$ so that $h(x_1)=+1$ implies $h(x_2)=+1.$
+%
+\end{itemize}
+%
+Therefore, we infer that $VC_1(\Hspace \cup \Hspace')< 3,$ as three points cannot be shattered by $\Hspace \cup \Hspace'.$
+%
+With this, we conclude that $VC_1(\Hspace \cup \Hspace')=2.$
+
+
+%
+\item Consider now the feature space $\Xspace = \{0,1\}^p$ for some $p\in \N$ and let 
+%
+$$\Hspace = \{ h_{t}:\Xspace \to \Yspace \, | \,t \in \{0,1,2,\ldots,p+1\} \}$$ 
+%
+be the hypothesis space of threshold classifiers on bitstrings, where $h_{t}(\xv) = 1$ for $\sum_{i=1}^p x_i \geq t$ and $=-1$ otherwise. 
+%
+Thus, instances are bitstrings of length $p$, and $h_{t}$ classifies an instance as positive if the number of 1s in the bitstring is at least $t,$ e.g., $h_3(0,1,1,0,0)=-1$ and $h_3(1,1,1,0,1)=+1$. 
+%
+What is $VC_p(\Hspace)?$
+
+\textbf{Solution:}
+%
+One arbitrary point $\xv \in \Xspace$ can be shattered since $h_{d+1}\equiv -1$ and $h_0\equiv1.$ 
+%
+Therefore, $VC_p(\Hspace)\geq 1.$
+
+Now define $N_1(\xv) = \# \{ x_j=1 ~|~ \xv=(x_1,\ldots,x_n) \}$, which denotes the number of ones in $ \xv \in \Xspace$.
+%
+If $N_1(\xv)=N_1(\xv')$, then $h_t(\xv)=h_t(\xv'), \forall \xv,\xv' \in \Xspace, t \in \{0,\ldots,p+1\}$, i.e., the ordering of the zeros resp.\ ones is not relevant, but only their total number. 
+%
+Thus, the ``interesting" candidate points are
+%
+\[ X_{cand} = \left\{ \begin{pmatrix}0 \\ 0 \\ \vdots \\ 0 \end{pmatrix}, \begin{pmatrix}1 \\ 0 \\ \vdots \\ 0 \end{pmatrix}, \begin{pmatrix}1 \\ 1 \\ \vdots \\ 0 \end{pmatrix}, \ldots, \begin{pmatrix}1 \\ 1\\ \vdots \\ 1 \end{pmatrix} \right\}. \]  
+
+But for each $\xv,\xv'\in X_{cand}$ it holds that $\exists~t : (h_t(\xv),h_t(\xv'))=(0,1)$, then $\forall t' \in \{0,\ldots,p+1\}~(h_{t'}(\xv),h_{t'}(\xv')) \neq (1,0) $.
+%
+We can show this indirectly, by assuming such a $t'$ exists and distinguish two cases:
+%
+\begin{enumerate}
+%
+	\item $t'<t$, then $h_t(\xv')=1 \implies h_{t'}(\xv')=1~$  
+%
+	\item $t'\geq t$, then $h_t(\xv)=0 \implies h_{t'}(\xv)=0~$ 
+%
+\end{enumerate} 
+%
+Both implications are contradictions.
+%
+It can be shown similarly: if for each $\xv,\xv'\in X_{cand}$ it holds that $\exists~t : (h_t(\xv)=h_t(\xv'))=(1,0)$, then $\forall t' \in \{0,\ldots,p+1\}~(h_t'(\xv)=h_t'(\xv')) \neq (0,1) $.
+%
+Hence, $VC_p(\Hspace)< 2$ as two points cannot be shattered by $\Hspace.$
+%
+In summary, we conclude that $VC_p(\Hspace) = 1.$
```

Comment:
Kann man hier noch eine Intuition herstellen, warum das so ist oder was das heißt?

Reply:
Ja, ich hätte in der Live Session darauf hingewiesen, dass oftmals gilt: VC-Dim eines Hypthesenraums = Anzahl an freien Pararmetern. Das gilt ja für alle Beispiele auf diesem Zettel und auch für die Beispiele, die in der VL kamen. Ich schreibe das einfach mal als Kommentar in die .Rnw Datei.